R/ptype-abbr-full.R

```diff
@@ -74,26 +81,55 @@ vec_ptype_full.default <- function(x, ...) {
 
 #' @export
 vec_ptype_abbr.default <- function(x, ...) {
-  if (is.object(x)) {
-    unname(abbreviate(vec_ptype_full(x), 8))
+  if (is.data.frame(x)) {
+    type <- class(x)[[1]]
+  } else if (is.object(x)) {
+    type <- vec_ptype_full(x)
   } else if (is_vector(x)) {
-    switch(typeof(x),
-      list = "list",
-      logical = "lgl",
-      integer = "int",
-      double = "dbl",
-      character = "chr",
-      complex = "cpl",
-      list = "list",
-      expression = "expr",
-      raw = "raw",
-      abbreviate(typeof(x))
-    )
+    type <- vec_ptype_abbr_bare(x, ...)
   } else {
     abort("Not a vector.")
   }
+  unname(abbreviate(type, 8))
 }
 
+vec_ptype_full_data_frame <- function(x, ...) {
```

Comment:
Are the dots useful for anything here? If not, can we remove them?

Reply:
I wrote these internal methods as if they were not internal.